### DISCUSSION TITLE
Collaborative learning == Superpowered learning

### DISCUSSION THREAD

**Discussion by PatrickAlphaC** [Cyfrin/foundry-full-course-cu#285] [Announcements]
I'm LOVING seeing so many people take the time to help each other out. Helping others is good for the world, but it also is a great way for YOU to learn!

When I was first getting started in web3, one of the ways I learned the fastest was trying to solve other people's problems. It gave me an opportunity to work on and try to learn from another angle, giving me more context for if/when I run into a similar problem. 

I tried to solve as many problems of others as I could, and it supercharged my learning 💪

Anyways, love to see it, and happy learning!

**Comment by GenevaRoc** [comment]
Hey Patrick! I started your course a week ago, I have no experience with any programming language so it's a bit difficult understanding the data types in solidity. Would you advice I read about solidity data types and then come back to your course or I should ignore and continue with the course?

**Reply by ghost** [reply]
Heya @GenevaRoc I'll share with you some tips I use to learn while taking up this course. 
- Make a note of every concept ( it can be in a notebook or [notion](https://www.notion.so/)).
- After completion of each lesson instead of moving to the next lesson I go through the notes I have taken and revise the concepts.
- I learn solidity first from the website [solidity-by-example](https://solidity-by-example.org/) and then for deeper understanding I either prefer asking with the AI or going through official [solidity documentation](https://docs.soliditylang.org/en/v0.8.20/).
- I do not go to the next lesson until I revise the lesson completed at least once. 
- Rewatch the concepts again and again until it gets stored in your mind permanently. (Make your fingers dirty... no not like that. I meant to say code more and more 😜)
- and yes do not jump into some other course just to learn Solidity. You'll end up in tutorial hell, which is the worst thing that can happen to you. (I know that since you have no experience in any programming language it will be a bit overwhelming. BELIEVE ME, I WAS TOO IN THAT SITUATION. But everything will be alright. Just make sure you understand each line of code you are writing before moving on to the next.)

**All the best** Let's learn together buddy 👍

**Reply by GenevaRoc** [reply]
Thank you!

**Reply by Mindsetter** [reply]
Very helpful

**Reply by Sean-Miningah** [reply]
Very helpful, just started off, thank you @edsondsouza
